prime-router/src/main/kotlin/SubmissionReceiver.kt

```diff
@@ -237,7 +237,8 @@ class ELRReceiver : SubmissionReceiver {
         val actionLogs = ActionLogger()
 
         // check that our input is valid HL7. Additional validation will happen at a later step
-        var messages = HL7Reader(actionLogs).getMessages(content)
+        val (messages, errors) = HL7Reader().getMessages(content)
```

Comment:
Why change this code. The actionLogger is being passed in. Revert.

Reply:
It decouples the HL7Reader from the Logging, allowing better single responsibility, limiting side effects, and increasing maintainability and testability. Check the consequences in the potential simplification of tests in particular.